### ISSUE TITLE
Bot token is displayed in the log file!

### ISSUE THREAD

**Issue by trevorjd** [GeheimagentNr1/DiscordIntegration#19]
**Versions**
Minecraft Version: [e.g. 1.16.5] 1.18.2
Minecraft Forge Version: [e.g. 36.1.4] 40.1.80
Mod Version: [e.g. 1.0.0] 2.2.0

**Log files**

```
[19Sep2022 22:25:50.774] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: Loading "Discord Integration" Server Config
[19Sep2022 22:25:50.774] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: [active] = true
[19Sep2022 22:25:50.774] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: [bot_token] = OTkyNjIxNjUzOREDACTEDREDACTEDREDACTED9f3eu6IUD53E
[19Sep2022 22:25:50.775] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: [channel_id] = 7614REDACTED3872
[19Sep2022 22:25:50.775] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: [command_prefix] = !
[19Sep2022 22:25:50.775] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: [other_bots, transmit_bot_messages] = false
[19Sep2022 22:25:50.775] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: [max_char_count] = -1
[19Sep2022 22:25:50.775] [Server thread/INFO] [de.geheimagentnr1.discordintegration.config.ServerConfig/]: [commands] = 
```

**Describe the bug**
The mod displays the bot token, which is supposed to be super-secret, in the log.
This is unnecessary and a terrible security risk since people frequently need to paste their logs to public places to get tech support.

**To Reproduce**
Steps to reproduce the behaviour:

1. Click on '....'
1. Scroll down to '....'
1. See error

**Expected behaviour**
Mod should not display private data in the log.

**Screenshots**

**Additional context**

┆Issue is synchronized with this [Trello card](https://trello.com/c/0QPhsArX) by [Unito](https://www.unito.io)

**Comment by Deliphin** [comment]
I know you don't update older versions anymore, but where this is a security issue, I really recommend backporting the fix for this to previous minecraft versions.

**Comment by GeheimagentNr1** [comment]
<table><tbody><tr><th>Status:</th><td>Released</td></tr><tr><th>Issue type:</th><td>New Feature</td></tr><tr><th>Fix type:</th><td>New feature</td></tr><tr><th>Fix version:</th><td>3.0.0</td></tr><tr><th>Fix release date:</th><td>12.11.2023</td></tr></tbody></table>

**TODOs:**

*   [x] Display only stars for the bot token in the logs

**Comment by GeheimagentNr1** [comment]
This feature was only implemented for version 3.0.0 and later, because the small number of bug reports with logs does not justify the effort.

Implemented with version 3.0.0 for Minecraft 1.18.1, 1.18.2, 1.19.2, 1.19.4 and 1.20.2
CurseForge: https://www.curseforge.com/minecraft/mc-mods/discord-integration
Modrith: https://modrinth.com/mod/discord-integration-(di)

**Comment by trevorjd** [comment]
Personally I feel it's a bigger problem than you may realise.  You may not receive many bug reports with logs but the modding community as an whole pretty much always demands logs and many unwary server owners may accidentally disclose their bot token when reporting an issue about a different mod.  But yeah, you have to draw a line in the sand somewhere.

Thanks very much for fixing this.  I'll be upgrading my servers ASAP!
